src/modules/ekf2/EKF/control.cpp

```diff
@@ -428,3 +429,12 @@ void Ekf::controlAuxVelFusion()
 		}
 	}
 }
+
+void Ekf::controlGravityFusion()
+{
+	// fuse gravity observation if our overall acceleration isn't too big
```

Comment:
Limit to when there's minimal manoeuvring?

eg `is_manoeuvre_level_high`

https://github.com/PX4/PX4-Autopilot/blob/32cab66c44b9ad950bd04802464f968fde439873/src/modules/ekf2/EKF/covariance.cpp#L121-L122

Reply:
I like that the current check _only_ depends on acceleration, which keeps it closely tied to the assumptions of the method itself. But I can't think of a  situation where the angular velocity is persistently high but the acceleration is nominal, which is where we'd really have problems.

I'll defer to y'all's judgment on this one.

Reply:
I think this should not run when we have GPS or some other form of position or velocity aiding that is measurement based.

Reply:
Make _control_status.flags.fake_pos == true a requirement for this to run

Reply:
You could also use `isHorizontalAidingActive()`.